DEBUG_FRAME_SIZE.md

```diff
@@ -0,0 +1,107 @@
+# Debugging Large Frame Size Issues
+
+## Problem
+When you encounter the error "The frame is bigger than expected. Length: 808922930, max: 1048576", it indicates that iOS devices are sending data packets larger than the default buffer limits.
+
+## Investigation Steps
+
+### 1. Enable Debug Logging
+Set these environment variables to investigate the issue:
+
+```bash
+export APPIUM_IOS_DEBUG_FRAME_SIZE=true
+export APPIUM_IOS_LOG_LARGE_FRAMES=true
+export APPIUM_IOS_MONITOR_MEMORY=true
+export APPIUM_IOS_LARGE_FRAME_THRESHOLD=10485760  # 10MB threshold
+```
+
+### 2. Run Your Tests
+With debug logging enabled, run your Appium tests. The enhanced logging will show:
+- Which service is causing the large frame (WebInspector, AFC, etc.)
+- Exact frame sizes being received
+- Memory usage at the time of the issue
+- Detailed error information
+
+### 3. Analyze the Logs
+Look for these log messages:
+```
+Large frame detected: 808922930 bytes (771.45 MB)
+Service: WebInspector, Max allowed: 1048576 bytes
+Memory usage - RSS: 245.67 MB, Heap: 123.45 MB
+```
+
+### 4. Identify the Root Cause
+Based on the service name and frame size, determine:
+- **WebInspector**: Usually indicates large JavaScript debugging data
+- **AFC**: File transfer operations with large files
+- **USBMUX**: Device communication issues
+- **Plist Service**: Large property list data
+
+## Common Causes
+
+### WebInspector Large Frames
+- **Cause**: JavaScript debugging data, DOM snapshots, or performance profiles
+- **Solution**: Consider disabling detailed debugging or using streaming
+
+### AFC Large Frames  
+- **Cause**: Large file transfers (logs, screenshots, app bundles)
+- **Solution**: Implement chunked file transfers
+
+### Memory Issues
+- **Cause**: Multiple parallel test sessions consuming too much memory
+- **Solution**: Reduce parallel sessions or implement memory cleanup
+
+## Temporary Workarounds
+
+### Option 1: Increase Limits (Not Recommended)
+```javascript
+// Only use this after investigation
+const MAX_FRAME_SIZE = 1024 * 1024 * 1024; // 1GB
```

Comment:
this is not very practical as such change would need to recompile the whole module. We may also have this limit configurable as environment variable